modules/hostdb/scan.go

```diff
@@ -0,0 +1,102 @@
+package hostdb
+
+// scan.go contians the functions which periodically scan the list of all hosts
+// to see which hosts are online or offline, and to get any updates to the
+// settings of the hosts.
+
+import (
+	"crypto/rand"
+	"math/big"
+	"time"
+
+	"github.com/NebulousLabs/Sia/build"
+	"github.com/NebulousLabs/Sia/modules"
+	"github.com/NebulousLabs/Sia/types"
+)
+
+const (
+	DefaultReliability  = 20
+	InactiveReliability = 10
+	UnreachablePenalty  = 1
+)
+
+// threadedProbeHost tries to fetch the settings of a host. If successful, the
+// host is put in the set of active hosts. If unsuccessful, the host id deleted
+// from the set of active hosts.
+//
+// TODO LOG: Log what happens in this function.
+func (hdb *HostDB) threadedProbeHost(entry *modules.HostEntry) {
+	// Request the most recent set of settings from the host.
+	var settings modules.HostSettings
+	err := hdb.gateway.RPC(entry.IPAddress, "HostSettings", func(conn modules.NetConn) error {
+		return conn.ReadObject(&settings, 1024)
+	})
+
+	// Now that network communicaiton is done, lock the hostdb.
+	id := hdb.mu.Lock()
+	defer hdb.mu.Unlock(id)
+	if err != nil {
+		// Beacuse there was an error, decrement the reliability.
+		entry.Reliability = entry.Reliability.Sub(types.NewCurrency64(UnreachablePenalty))
+
+		// If the reliability has fallen below InactiveReliability, remove the host from the list
+		// of active hosts.
+		node, exists := hdb.activeHosts[entry.IPAddress]
+		if exists && entry.Reliability.Cmp(types.NewCurrency64(InactiveReliability)) < 0 {
+			delete(hdb.activeHosts, entry.IPAddress)
+			node.removeNode()
+		}
```

Comment:
maybe factor this out into an `hdb.removeNode(entry.IPAddress)`, which would be a no-op if the entry didn't exist.